### ISSUE TITLE
a total_loss issue in voxel_only stage of training?

### ISSUE THREAD

**Issue by dmitrykudinov** [facebookresearch/meshrcnn#86]
Hi @gkioxari ,

It looks like there might be an issue in the way the total loss is calculated in shapenet's implementation at the early iterations while only the voxel_head is trained.

[Here](https://github.com/facebookresearch/meshrcnn/blob/master/tools/train_net_shapenet.py#L191) we see the partial non-voxel losses are being dropped from the dict. But the `total_loss` used in backprop still contains the chamfer, norm and edge losses calculated on dummy meshes leading to a noisy convergence for the voxel_only iters. I could not find something like `loss = loss * 0. + losses["voxel"] * loss_fn.voxel_weight` in the associated codepath of train_net_shapenet... Am I missing something here, or is it truly an issue?

**Comment by gkioxari** [comment]
Oh shoot! I somehow missed this. I am so sorry. It's been some busy couple of weeks due to CVPR 2021. 

The intended behavior is for the mesh refinement branch to not train at all during the voxel_only iterations, which are the first 200 iterations or so. The reason we set this is because initially the voxel predictions are bad, leading to cubified meshes which are huge and nonsense and take too much memory for no reason (sometimes OOM too). So we do some warm up training for the voxel branch and use a dummy mesh. Looking at the code, I agree with you that the mesh loss is actually not zero-ed out properly (I will blame Justin for that :P). I will fix it when I get the time. But in practice, even though it's not zero-ed out, it should have no effect really to the model and its final state and convergence.

**Comment by dmitrykudinov** [comment]
Sounds good, closing this one. 
Good luck at CVPR! Have you published any related drafts (arxiv?) yet? Always curious to see what you guys are coming up with this time.

**Comment by gkioxari** [comment]
I am a program co-chair for CVPR so I can't submit anything to it! But some arxiv papers are coming out soon! I will add a fix to the issue you brought up soon :)
